stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py

```diff
@@ -0,0 +1,280 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Francesco Apruzzese
+#    Copyright 2015 Apulia Software srl
+#    Copyright 2015 Lorenzo Battistini - Agile Business Group
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api, exceptions, _
+import openerp.addons.decimal_precision as dp
+from datetime import datetime
+
+
+class StockPickingPackagePreparationLine(models.Model):
+
+    _name = 'stock.picking.package.preparation.line'
+    _description = 'Package Preparation Line'
+    _order = 'sequence asc'
+
+    package_preparation_id = fields.Many2one(
+        'stock.picking.package.preparation', string='Stock Move',
+        ondelete='cascade')
+    name = fields.Text(string='Description', required=True)
+    move_id = fields.Many2one('stock.move', string='Stock Move',
+                              ondelete='cascade')
+    product_id = fields.Many2one('product.product', string='Product')
+    product_uom_qty = fields.Float(
+        digits_compute=dp.get_precision('Product Unit of Measure'),
+        help="If you change this quantity for a 'ready' picking, the system "
+             "will not generate a back order, but will just deliver the new "
+             "quantity")
+    product_uom = fields.Many2one('product.uom')
```

Comment:
`product_uom_id` - all relation fields should end in _id or _ids, depending on whether they are x2m (_ids) or x2o (_id).

This also needs a string attribute